ui/components/app/signature-request-original/signature-request-original.container.js

```diff
@@ -29,6 +30,20 @@ function mapStateToProps(state, ownProps) {
     from,
   );
   const isLedgerWallet = isAddressLedger(state, from);
+  const {
+    unapprovedTypedMessagesCount,
+    unapprovedPersonalMsgCount,
+    unapprovedDecryptMsgCount,
+    unapprovedEncryptionPublicKeyMsgCount,
+    unapprovedMsgCount,
+  } = state.metamask;
+  const messagesList = unconfirmedMessagesHashSelector(state);
+  const messagesCount =
+    unapprovedTypedMessagesCount +
+    unapprovedPersonalMsgCount +
+    unapprovedDecryptMsgCount +
+    unapprovedEncryptionPublicKeyMsgCount +
+    unapprovedMsgCount;
```

Comment:
Should we make a selector that does this aggregate message count? `getTotalUnapprovedMessagesCount`?

Reply:
@adonesky1 makes sense. Done. :)